parity/rpc_apis.rs

```diff
@@ -322,8 +322,9 @@ impl FullDependencies {
 				}
 				Api::EthPubSub => {
 					if !for_generic_pubsub {
-						let client =
+						let mut client =
 							EthPubSubClient::new(self.client.clone(), self.executor.clone());
+						client.add_sync_notifier(self.sync.sync_notification());
```

Comment:
Ok, it seems we have different implementations depending on `full/light` node. We could either have different constructors or maybe attempt to avoid repetition of the entire code that adds the handler and for instannce just pass a closure that takes `QueueInfo` and dispatches the notification (unless there is an abstraction for queue info already then we should just use that and make `new` generic)